http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/SdkEventLoopGroup.java

```diff
@@ -93,6 +111,13 @@ public ChannelFactory<? extends Channel> channelFactory() {
         return channelFactory;
     }
 
+    /**
+     * @return the {@link ChannelFactory} to be used with Netty Http Client.
+     */
+    public Class<? extends DatagramChannel> channelType() {
+        return channelType;
+    }
```

Comment:
Should we do `ChannelFactory<? extends DatagramChannel> datagramChannelFactory()` to be symmetric with the other channel factory? We'd use that type everywhere in here.

It would also avoid the overhead of the `ReflectiveChannelFactory`'s reflection whenever channels are created.

Reply:
That wouldn't be possible, I tried but as an example, `NioSocketChannel::new` doesn't satisfy the type `ChannelFactory<? extends DatagramChannel>`.

Reply:
Oh, now I see what you meant. Alright, changes pushed.